components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -300,6 +302,32 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
         return geckoResult
     }
 
+    override fun onSharePrompt(
+        session: GeckoSession,
+        prompt: PromptDelegate.SharePrompt
+    ): GeckoResult<PromptResponse> {
+        val geckoResult = GeckoResult<PromptResponse>()
+        val onSuccess = { geckoResult.complete(prompt.confirm(GECKO_PROMPT_SHARE_RESULT.SUCCESS)) }
+        val onFailure = { geckoResult.complete(prompt.confirm(GECKO_PROMPT_SHARE_RESULT.FAILURE)) }
+        val onDismiss = { geckoResult.complete(prompt.dismiss()) }
+
+        geckoEngineSession.notifyObservers {
+            onPromptRequest(
+                PromptRequest.Share(
+                    ShareData(
+                        title = prompt.title,
```

Comment:
If all values are null, since they're all nullable, what are we sharing then? Should we just dismiss immediately?

Reply:
Theoretically Gecko shouldn't pass in a request if everything is null, and will [throw a TypeError instead](https://w3c.github.io/web-share/#navigator-interface). We don't have an API to throw that error from here so I don't think we need to double-check.

Reply:
Did we communicate this issue to the GV team? This way we can enforce this restriction to the type level, they can improve the API to make its values non-null.

Reply:
> Did we communicate this issue to the GV team? This way we can enforce this restriction to the type level, they can improve the API to make its values non-null.

I looked at the spec, and all fields are optional, so there's no way for GV to enforce non-null at the GV API surface. 🤷‍♂